pkg/controller/backup_finalizer_controller.go

```diff
@@ -228,6 +222,19 @@ func (r *backupFinalizerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 func (r *backupFinalizerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&velerov1api.Backup{}).
+		WithEventFilter(predicate.Funcs{
+			// Only process backups that are updated to finalizing phase after plugin operations are complete
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				backup, ok := e.ObjectNew.(*velerov1api.Backup)
+				if !ok {
+					return false
+				}
+				return backup.Status.Phase == velerov1api.BackupPhaseFinalizing || backup.Status.Phase == velerov1api.BackupPhaseFinalizingPartiallyFailed
```

Comment:
We should return the opposite result here.
Restore reconciler should also be changed.

Reply:
```go
	// Update returns true if the Update event should be processed
	UpdateFunc func(event.UpdateEvent) bool
```

Don't we want events to be processed if CR was updated to finalizing?

Reply:
@kaovilai so there are 2 cases where we want to process backups in this controller:
1) backups updated to Finalizing or FinalizingPartiallyFailed phase
2) backups already in finalizing phase that returned err from prior reconcile (i.e. requeueing)

Does this predicate change affect requeue behavior?